components/suggestion/switch_channel_provider.jsx

```diff
@@ -608,16 +643,39 @@ export default class SwitchChannelProvider extends Provider {
                     channel,
                 );
             }
-            wrappedChannel.type = Constants.MENTION_UNREAD_CHANNELS;
-            channels.push(wrappedChannel);
+            wrappedChannel.type = channelType;
+            channelList.push(wrappedChannel);
         }
+        return channelList;
+    }
 
-        const channelNames = channels.map((wrappedChannel) => wrappedChannel.channel.id);
+    async fetchChannels(resultsCallback, size = 20) {
+        const state = getState();
+        const teamId = getCurrentTeamId(state);
+        if (!teamId) {
+            return;
+        }
+        const channelsAsync = getChannels(teamId, 0, size)(store.dispatch, store.getState);
+        let channels;
+
+        try {
+            const {data} = await channelsAsync;
+            channels = data;
+        } catch (err) {
+            store.dispatch(logError(err));
+        }
+
+        if (this.latestPrefix !== '') {
+            return;
+        }
+        const sortedChannels = this.wrapChannels(channels, Constants.MENTION_PUBLIC_CHANNELS).
+            sort(sortChannelsByRecencyAndTypeAndDisplayName).slice(0, 20);
```

Comment:
I think we should be fine to use `sortChannelsByTypeAndDisplayName` here. If the user hasn't viewed any other channels yet, the recency sorting won't do anything.

Reply:
Yeah make sense have removed it